libs/langchain/langchain/chat_models/fake.py

```diff
@@ -53,7 +53,17 @@ def _stream(
         for c in response:
             if self.sleep is not None:
                 time.sleep(self.sleep)
-            yield ChatGenerationChunk(message=AIMessageChunk(content=c))
+            if isinstance(response, dict):
```

Comment:
should this be checking if response is a dict or if c is a dict? if the former, does it need to be in the for loop?

Reply:
Yes, good catch! This really needs to be in the `_generate` method. Can you take a look at the latest version? I will add tests soon.